numpy/f2py/f90mod_rules.py

```diff
@@ -110,11 +110,22 @@ def dadd(line, s=doc):
                 notvars.append(b['name'])
         for n in m['vars'].keys():
             var = m['vars'][n]
-            if (n not in notvars) and (not l_or(isintent_hide, isprivate)(var)):
+            # heuristic to find public/private declarations of filtered
+            # subroutines
+            if len(var) == 1 and 'attrspec' in var and var['attrspec'][0] in ('public', 'private'):
+                is_not_var = True
```

Comment:
I wonder if this entire check could be extracted into a functional form in `auxfuncs.py` instead. Similar to the existing:

```python
def isprivate(var):
    return 'attrspec' in var and 'private' in var['attrspec']
```

This is a bit nitpicky and mostly for consistency with the rest of the F2PY code, the form here also works.

Reply:
Also, what about the `protected` access property? (not sure how / if this is handled in F2PY now actually).

Reply:
As far as I can see, the `protected` property is correctly handled, i.e., using it produced no errors for me. However, I'm not familiar enough with it to tell if I missed anything here.